contracts/clearing/TradingApp.sol

```diff
@@ -113,53 +113,38 @@ contract TradingApp is IForceMoveApp {
 		ITradingTypes.Settlement memory settlement,
 		RecoveredVariablePart[] calldata proof
 	) internal pure {
+		VariablePart memory currProof = proof[0].variablePart;
+		VariablePart memory nextProof = proof[1].variablePart;
+		require(
+			currProof.turnNum >= 2 && nextProof.turnNum >= 2,
+			'only prefund and postfund can have turn number < 2'
+		);
+
 		bytes32[] memory orderIDs = new bytes32[](proof.length);
-		for (uint256 i = 0; i < proof.length - 1; i++) {
-			VariablePart memory currProof = proof[i].variablePart;
-			VariablePart memory nextProof = proof[i + 1].variablePart;
+		uint256 prevTurnNum = 0;
+		for (uint256 i = 0; i < proof.length - 1; i += 2) {
+			currProof = proof[i].variablePart;
+			nextProof = proof[i + 1].variablePart;
 
-			require(
-				currProof.turnNum >= 2 && nextProof.turnNum >= 2,
-				'only prefund and postfund can have turnNum < 2'
-			);
+			require(prevTurnNum + 1 == currProof.turnNum, 'turns are not consecutive');
 			require(currProof.turnNum + 1 == nextProof.turnNum, 'turns are not consecutive');
 
 			// Verify validity of orders and responses
-			if (i % 2 == 0) {
-				// If current proof contains an order,
-				// then the next one must contain a response
-				// with the same order ID
-				ITradingTypes.Order memory order = abi.decode(
-					currProof.appData,
-					(ITradingTypes.Order)
-				);
-				ITradingTypes.OrderResponse memory orderResponse = abi.decode(
-					nextProof.appData,
-					(ITradingTypes.OrderResponse)
-				);
-				require(
-					orderResponse.orderID == order.orderID,
-					'orderResponse.orderID != order.orderID'
-				);
-				orderIDs[i] = order.orderID;
-			} else {
-				// If current proof contains a response,
-				// then the next one must be an order
-				// with a different order ID, since they are not related
-				ITradingTypes.OrderResponse memory orderResponse = abi.decode(
-					currProof.appData,
-					(ITradingTypes.OrderResponse)
-				);
-				ITradingTypes.Order memory order = abi.decode(
-					nextProof.appData,
-					(ITradingTypes.Order)
-				);
-				require(
-					order.orderID != orderResponse.orderID,
-					'order.orderID == orderResponse.orderID'
-				);
-				orderIDs[i] = order.orderID;
-			}
+			ITradingTypes.Order memory order = abi.decode(currProof.appData, (ITradingTypes.Order));
+			ITradingTypes.OrderResponse memory orderResponse = abi.decode(
+				nextProof.appData,
+				(ITradingTypes.OrderResponse)
+			);
+
+			// If current proof contains an order,
+			// then the next one must contain a response
+			// with the same order ID
+			require(
+				orderResponse.orderID == order.orderID,
+				'order and response orderIDs do not match'
+			);
+			orderIDs[i] = order.orderID;
```

Comment:
Let's rename `orderIDs` to `proofDataHashes` or similar, and store `keccak256(currProof.appData)` and `keccak256(nextProof.appData)` there as agreed.